Makefile

```
@@ -123,43 +134,36 @@ $(PROTOC_GEN_DEEPCOPY):
 $(KIND):
 	@GOBIN=$(LOCALBIN) go install -mod=readonly sigs.k8s.io/kind@v0.26.0
 
+$(CONTROLLER_GEN):
+	GOBIN=$(LOCALBIN) go install -mod=readonly sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_TOOLS_VERSION)
+
 .PHONY: clean
-clean: 
+clean:
 	@rm -rf $(LOCALBIN) $(PROJECT_DIR)/$(OUT_DIR)
 
 ##@ Code Gen
 
 $(PROTOBUF_GEN): $(PROTOBUF_API_SRC) $(PROTOC) $(PROTOC_GEN_GO) $(PROTOC_GEN_GRPC) $(PROTOC_GEN_DEEPCOPY) ## Generates Go files from protobuf-based API files
 	@PATH=$(LOCALBIN):$$PATH $(PROTOC) --proto_path=$(PROTOBUF_API_DIR) --go_out=$(API_GEN_DIR) --go-grpc_out=$(API_GEN_DIR) --golang-deepcopy_out=:$(API_GEN_DIR) $(PROTOBUF_API_DIR)/**/*.proto
 
+$(DEEP_COPY_GEN): $(DEEP_COPY_SRC) ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
+	$(CONTROLLER_GEN) object:headerFile=$(LICENSE_FILE) paths="./..."
+
+$(CRD_GEN): $(CRD_SRC) ## Generate CustomResourceDefinition objects.
+	$(CONTROLLER_GEN) crd paths="./..." output:crd:artifacts:config=chart/crds
+
```

Moving forward we will also add webhooks, so having the entire generation as one command will be more efficient.

What we missed here is dependency to the `controller-gen` itself - if it's not present in the path it will now be automatically fetched.

```suggestion
$(CRD_GEN): $(CRD_SRC) $(CONTROLLER_GEN) ## Generates CRDs and other controller-runtime scaffolding.
	$(CONTROLLER_GEN) paths="$(CRD_SRC_DIR)/..." \
		crd output:crd:artifacts:config="$(CRD_GEN_DIR)" \
		object:headerFile="$(LICENSE_FILE)"
```

---

There's more to that btw (RBAC generation, kube-builder directives in controllers etc), but we can fine-tune it as we go.

---

One other thing from the top of my head - if someone opens PR with code changes but forgets to regenerate CRDs and other stuff we will end up with misaligned artifacts, i.e. YAMLs will be behind. I will capture it as an issue.

---

Done.